group1/comm/models.py

```diff
@@ -32,3 +32,13 @@ class UserRoom(models.Model):
 
 	class Meta:
 		unique_together = ('user', 'room')
+
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+@receiver(post_save, sender=Message)
+def delete_oldest(sender, instance, **kwargs):
+	curroom = instance.room
+	while (Message.objects.filter(room = curroom).count() > 5000):
```

Comment:
this is a really good first step. this query pulls in all room messages into memory, and then issues delete queries and each message after the 5000th. who knows how large a message is ... ? ;) 
it would be great if you could issue one "query" to delete all messages > 5000, rather than pulling in all the room messages into memory and then issuing a delete query per message. 

that would make it cleaner and reduce "load" on the DB (LOL, i know), as well as reduce the memory footprint on the webapp. this is an edge-case, but i think it be a worthwhile exercise learning how to do that. ;)